nsxt/resource_nsxt_transport_node.go

```diff
@@ -0,0 +1,1987 @@
+				"static_ip_list": {
+					Type:        schema.TypeList,
+					MaxItems:    1,
+					Optional:    true,
+					Description: "IP assignment specification for Static IP List.",
+					Elem: &schema.Resource{
+						Schema: map[string]*schema.Schema{
+							"default_gateway": {
+								Type:         schema.TypeString,
+								Required:     true,
+								Description:  "Gateway IP",
+								ValidateFunc: validateSingleIP(),
+							},
+							"ip_addresses": {
+								Type:        schema.TypeList,
+								Description: "List of IPs for transport node host switch virtual tunnel endpoints",
+								MinItems:    1,
+								Required:    true,
+								Elem: &schema.Schema{
+									Type:         schema.TypeString,
+									ValidateFunc: validateSingleIP(),
+								},
+							},
+							"subnet_mask": {
+								Type:         schema.TypeString,
+								Required:     true,
+								Description:  "Subnet mask",
+								ValidateFunc: validateSingleIP(),
+							},
+						},
+					},
+				},
+				"static_ip_mac_list": {
+					Type:        schema.TypeList,
+					MaxItems:    1,
+					Optional:    true,
+					Description: "IP and MAC assignment specification for Static IP List",
+					Elem: &schema.Resource{
+						Schema: map[string]*schema.Schema{
+							"default_gateway": {
+								Type:         schema.TypeString,
+								Required:     true,
+								Description:  "Gateway IP",
+								ValidateFunc: validateSingleIP(),
+							},
+							"ip_mac_list": {
+								Type:        schema.TypeList,
+								Description: "List of IPs and MACs for transport node host switch virtual tunnel endpoints",
+								MinItems:    1,
+								Required:    true,
+								Elem: &schema.Resource{
+									Schema: map[string]*schema.Schema{
+										"ip": {
+											Type:         schema.TypeString,
+											Required:     true,
+											Description:  "IP address",
+											ValidateFunc: validateSingleIP(),
+										},
+										"mac": {
+											Type:         schema.TypeString,
+											Optional:     true,
+											Description:  "MAC address",
+											ValidateFunc: validation.IsMACAddress,
+										},
+									},
+								},
+							},
+							"subnet_mask": {
+								Type:         schema.TypeString,
+								Required:     true,
+								Description:  "Subnet mask",
+								ValidateFunc: validateSingleIP(),
+							},
+						},
+					},
+				},
+				"static_ip_pool_id": {
+					Type:        schema.TypeString,
+					Optional:    true,
+					Description: "IP assignment specification for Static IP Pool",
+				},
+			},
+		},
+	}
+}
+
+func getTransportNodeFromSchema(d *schema.ResourceData) (*model.TransportNode, error) {
+	description := d.Get("description").(string)
+	displayName := d.Get("display_name").(string)
+	tags := getMPTagsFromSchema(d)
+	failureDomain := d.Get("failure_domain").(string)
+	hostSwitchSpec, err := getHostSwitchSpecFromSchema(d)
+	if err != nil {
+		return nil, handleCreateError("TransportNode", "", err)
```

Comment:
`handleCreateError` (which translates to `logAPIError`) is based on policy model error structure. We probably want to code similar functions (hopefully with code reuse) to handle MP model error structure. This can be handled in a separate PR.